src/prefect/new_flow_engine.py

```diff
@@ -47,14 +51,18 @@
 
 @dataclass
 class FlowRunEngine(Generic[P, R]):
-    flow: Flow[P, Coroutine[Any, Any, R]]
+    flow: Optional[Flow[P, Coroutine[Any, Any, R]]] = None
     parameters: Optional[Dict[str, Any]] = None
     flow_run: Optional[FlowRun] = None
+    flow_run_id: Optional[UUID] = None
     _is_started: bool = False
     _client: Optional[PrefectClient] = None
     short_circuit: bool = False
 
     def __post_init__(self):
+        if self.flow is None and self.flow_run_id is None:
+            raise ValueError("Either a flow or a flow_run_id must be provided.")
```

Comment:
Should we also raise if a `flow` and `flow_run_id` are provided?

Reply:
Hmmmm it's an interesting question; there's no known reason _not_ to, but I'm usually hesitant to add a constraint without context for the failure mode I'm preventing.  Said differently, it's not clear to me there's necessarily a problem in that situation, as things may still run as expected - without knowing how the situation arises I don't feel I have enough information to say one way or the other and therefore I'd rather not add it preemptively.

Reply:
That's fair! Thinking of future developers, we may want to document that a provided `flow` will override a provided `flow_run_id`, once we get around to adding dcostrings 😄